app/src/main/java/nl/tudelft/b_b_w/view/AddBlockActivity.java

```diff
@@ -70,11 +70,15 @@ public void onAddBlock(View view) {
             String senderHash = senderHashText.getText().toString();
             String senderPublicKey = senderPublicKeyText.getText().toString();
 
+            //URGENCY: THIS NEED TO BE A SEPARATE INPUT FIELD FOR IBAN NUMBER
+            String senderIban = senderPublicKeyText.getText().toString();
+
+
 
             // create and add the block
             Block previous = handler.getLatestBlock(ownerName);
-
-            ConversionController conversionController = new ConversionController(ownerName, senderPublicKey, previous.getOwnHash(), senderHash);
+            int blockSeqNumber = previous.getSequenceNumber()+ 1;
+            ConversionController conversionController = new ConversionController(ownerName, blockSeqNumber, senderPublicKey, previous.getOwnHash(), senderHash, senderIban);
             String ownHash = conversionController.hashKey();
             Block block = new Block(
```

Comment:
#15 is a whole another big issue, theres no quick fix. It should be fixed all at once as this kind of error are in another files that I have not touched too.